### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by tomermoshe**

### CODE REVIEW

File: bs-config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (eitanfr) [inline bs-config.json:1]: move it inside the mock-server folder

File: src/app/chat/chat-view/chat-view.component.ts [hunk 2]
```diff
@@ -31,6 +32,13 @@ export class ChatViewComponent implements OnInit, OnDestroy {
               private cd: ChangeDetectorRef) {
   }
 
+  sendMessageButtonFocus(event) {
+    this.messageInput.getElementRef().nativeElement.children[0].focus();
+    event.preventDefault();
```
Comment (eitanfr) [inline src/app/chat/chat-view/chat-view.component.ts:37]: does all event.stop are needed?

File: src/app/shared/services/authentication.service.ts [hunk 3]
```diff
@@ -21,25 +19,40 @@ export class AuthenticationService {
...
 
   async login(username: string, password: string): Promise<any> {
-    return this.accountsClient.loginWithPassword(username, password);
+    await this.accountsClient.loginWithPassword(username, password);
+    await this.setAuthMiddlewareToken();
+    return;
   }
 
   async logout() {
+    AuthorizationMiddleware.removeToken();
```
Comment (eitanfr) [inline src/app/shared/services/authentication.service.ts:55]: 👍

File: bs-config.json (hunk 1 above)
Comment (tomermoshe) [reply bs-config.json:1]: talked

File: src/app/chat/chat-view/chat-view.component.ts (hunk 2 above)
Comment (tomermoshe) [reply src/app/chat/chat-view/chat-view.component.ts:37]: yes

File: src/app/shared/services/authentication.service.ts (hunk 3 above)
Comment (tomermoshe) [reply src/app/shared/services/authentication.service.ts:55]: kentak

**Pull request merged**